scripts/generate-test-services.ts

```diff
@@ -103,68 +103,77 @@ function removeImports(str: string): string {
 function adjustRequestBuilder(str: string): string {
   return str
     .replace(
-      'return new CommonEntityRequestBuilder()',
+      /return new .*RequestBuilder\(\)/,
       "throw new Error('not implemented')"
     )
-    .replace('CommonEntityRequestBuilder', 'any');
+    .replace(
+      /static requestBuilder\(\):.*RequestBuilder/,
+      'static requestBuilder(): any'
+    );
 }
 
 function adjustCustomField(str: string): string {
   return str.replace(
-    'return Entity.customFieldSelector(fieldName, CommonEntity)',
-    'return new CustomField(fieldName,CommonEntity)'
+    'return Entity.customFieldSelector',
+    'return new CustomField'
   );
 }
 
 function addODataVersion(str: string): string {
   const firstProperty = "static _entityName = 'A_CommonEntity';";
+  const nameString = str.match(/static _entityName =.*/)![0];
   return str.replace(
-    firstProperty,
-    [firstProperty, 'readonly _oDataVersion: any;'].join(unixEOL)
+    nameString,
+    [nameString, 'readonly _oDataVersion: any;'].join(unixEOL)
   );
 }
 
 function removeJsDoc(str: string): string {
   return str.replace(/\/\*\*\n(?:\s+\*\s+.+\n)+\s+\*\/\n/g, '');
 }
 
-async function generateCommonTestEntity() {
-  const entity = await promises.readFile(
-    join(packageOutputDir, 'v2', 'common-service', 'CommonEntity.ts'),
-    'utf8'
-  );
-  const complexType = await promises.readFile(
-    join(packageOutputDir, 'v2', 'common-service', 'CommonComplexType.ts'),
-    'utf8'
+function readClasses(): Promise<string[]> {
+  const basePath = join(packageOutputDir, 'v4', 'common-service');
+  return Promise.all(
+    [
+      'CommonEntity.ts',
+      'CommonEntitySingleLink.ts',
+      'CommonComplexType.ts',
+      'NestedComplexType.ts'
+    ].map(name => promises.readFile(join(basePath, name), 'utf8'))
   );
+}
 
-  const adjustedEntity = [
-    removeImports,
-    removeJsDoc,
-    adjustRequestBuilder,
-    adjustCustomField,
-    addODataVersion
-  ].reduce((colllected, fn) => fn(colllected), entity);
-  const adjustedComplex = [removeImports, removeJsDoc].reduce(
-    (colllected, current) => current(colllected),
-    complexType
-  );
-  const adjusted = [
+async function generateCommonTestEntity() {
+  const files = await readClasses();
+
+  const [entity, entityLink, complexType, nestedComplex] = files
+    .map(str => removeImports(str))
+    .map(str => removeJsDoc(str));
+
+  const [adjustedEntity, adjustedEntityLink] = [entity, entityLink]
+    .map(str => adjustRequestBuilder(str))
+    .map(str => adjustCustomField(str))
+    .map(str => addODataVersion(str));
```

Comment:
Same here (then you would just iterate once).

Reply:
see comment below about nesting function calls

Reply:
I don't know which comment you are referring to, but this was supposed to just be a trigger for thought.